cumulus/pallets/xcmp-queue/src/migration.rs

```diff
@@ -48,6 +50,12 @@ impl<T: Config> OnRuntimeUpgrade for MigrationToV3<T> {
 			weight.saturating_accrue(T::DbWeight::get().writes(1));
 		}
 
+		if StorageVersion::get::<Pallet<T>>() == 3 {
+			weight.saturating_accrue(migrate_to_v4::<T>());
+			StorageVersion::new(4).put::<Pallet<T>>();
+			weight.saturating_accrue(T::DbWeight::get().writes(1));
```

Comment:
The struct is called `MigrateToV3`, so i think it should not migrate to V4.

Reply:
Changed it to `MigrateToV4`. Hope that's ok.

Reply:
Moved each migration to a different struct as suggested in a separate comment.